flink-tests/src/test/java/org/apache/flink/test/runtime/BlockingShuffleITCase.java

```diff
@@ -94,6 +95,7 @@ private JobGraph createJobGraph(int numRecordsToSend) {
         StreamGraph streamGraph = env.getStreamGraph();
         streamGraph.setGlobalDataExchangeMode(GlobalDataExchangeMode.ALL_EDGES_BLOCKING);
         streamGraph.setScheduleMode(ScheduleMode.LAZY_FROM_SOURCES);
+        streamGraph.setJobType(JobType.BATCH);
```

Comment:
hmm...maybe we could save ourselves some trouble by implicitly setting the job type if certain exchange modes / schedule modes are set.

In any case a comment would be good to clarify which of the 2 options (or both) results in the job being considered a batch one.

Reply:
I'm not sure if implicitly setting the job type is a good idea: How would you handle conflicting job types (e.g. StreamGraph.jobType = Streaming + StreamGraph.globalDataExchange = BLOCKING).
If we see that a lot more tests require setting the job type manually, then I'd revisit this.

Reply:
Yes, it was just a thought.